### DISCUSSION TITLE
tag good pratice

### DISCUSSION THREAD

**Discussion by joe97228** [community/community#21712] [New to GitHub]
<p>Hello everyone,</p>
<p>i’m a fullstack developper working for a compagny which passed all their projects Svn into Git two or three years ago.</p>
<p>We are doing our best to work thank to the gitFlow.</p>
<p>I have a big issue which confusing me, the tech-leads expecting us to tag  branches wich are not MASTER in order to deploy them in the production then doing a rebase .</p>
<p>As far as I know regarding to the git and gitFlow documentations the one branch we should tag is MASTER.</p>
<p>I can make a mistake so I’ll be more then thankfull from you guys to give me returns.</p>
<p>Regards<br>
Joe</p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/joe97228">@joe97228</a>,</p>
<p>Welcome to the Github Community Forum!</p>
<p>Yes, you are totally right.</p>
<p>You can let read this post to your tech-leads: <a href="https://nvie.com/posts/a-successful-git-branching-model/" rel="nofollow noopener">a successful git branching model</a></p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>

**Comment by joe97228** [comment]
<p>thank you <a class="mention" href="/wabri">@wabri</a>  :slight_smile:</p>

**Comment by crichID** [comment]
<p>Hi <a class="mention" href="/joe97228">@joe97228</a> </p>
<p>You can actually tag any branch in Git and GitHub. A tag is simply a pointer to a commit. Fun fact: a branch is also just a pointer to a commit. In the case of a branch, it is a pointer to the most recent commit in a line of work. In the case of a tag, it is a pointer to a specific point in time.</p>
<p>With regards to what your tech-leads are asking - deploying from something other than master is pretty common. For example, at GitHub we actually deploy from the feature branch and then merge the branch to master. For a really interesting read on how we do it, check out this post: <a href="https://github.blog/2015-06-02-deploying-branches-to-github-com/" rel="nofollow noopener">https://github.blog/2015-06-02-deploying-branches-to-github-com/</a></p>

**Comment by Wabri** [comment]
<p>The question is not if it is possible, but if is the right method to version. Is obvious that you can tag every branch. Is like create a new branch, you can create everywhere without problems.</p>
<p>The good conduct is to use the master only for tag a new version, git workflow is this. If Github follow another workflow to version is another question.</p>
<blockquote>
<p><a class="mention" href="/crichid">@crichid</a> wrote:</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>For a really interesting read on how we do it, check out this post: <a href="https://github.blog/2015-06-02-deploying-branches-to-github-com/" rel="nofollow noopener">https://github.blog/2015-06-02-deploying-branches-to-github-com/</a></p>
</blockquote>
<p>Thanks a lot to share this blog post :slight_smile:</p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>

**Comment by MChevy422** [comment]
<p><strong>Hello <span><a class="mention" href="/joe97228">@joe97228</a></span> ,</strong></p>
<p><strong>This post was moved to a different board that fits your topic of discussion a bit better. This means you’ll get better engagement on your post, and it keeps our Community organized so users can more easily find information.</strong></p>
<p><strong>As you’ll notice, your Topic is now in the How to use Git and Github board. No action is needed on your part; you can continue the conversation as normal here.</strong></p>
<p><strong>Let me know if you have any other questions or if I can help with anything else.</strong></p>
<p><strong>Greatly appreciated,</strong></p>
<p><strong>-MChevy422</strong></p>
